src/errors/logger.ts

```diff
@@ -1,5 +1,5 @@
-import * as fs from 'fs/promises'
-import {dirname} from 'path'
+import {mkdir, appendFile} from 'node:fs/promises'
+import {dirname} from 'node:path'
 import stripAnsi = require('strip-ansi')
```

Comment:
for some reason I though you'd need to get rid of `require` statements for ESM

Reply:
...so v3 enables ESM for clis/plugins but isn't itself ESM, right?  When does that need to happen?

Reply:
@mshanemc Correct. V3's major feature is that it fully supports CJS + ESM interoperability. The package itself is not ESM though.

Keeping it CJS for now allows CJS plugins to use v3, which makes the migration path for us and others much easier. I'm open to reconsidering that position though.